### ISSUE TITLE
HEAT OF FORMATION differs according to MOPAC version

### ISSUE THREAD

**Issue by YGuangye** [openmopac/mopac#231]
I have run calculations on the same system with same keywords using different versions (commercial MOPAC2016 version 19.350L and MOPAC v23.0.2 Linux) and obtained different result shown below. The deference in `HEAT OF FORMATION` is as much as 697.71 kcal/mol, which is about half of the value printed in the `H-BOND ENERGY` line. What causes this difference?  
Thanks in advance.

`Empirical Formula: C1039 H1617 N285 O327 S14 Cl  =  3283 atoms`

keywords:
`PL PM6-D3H4X MOZYME EPS=78.4 NSPA=60 1SCF LBFGS ITER EIGS GRADIENTS DISP`

MOPAC2016 version 19.350L
```
     SCF FIELD WAS ACHIEVED                                   

          HEAT OF FORMATION       =     -14156.30868 KCAL/MOL =  -59229.99554 KJ/MOL

          TOTAL ENERGY            =   -6884708.02037 KCAL/MOL
          ENERGY OF ATOMS         =    6870317.27255 KCAL/MOL
                              SUM =     -14390.74782 KCAL/MOL
          DISPERSION ENERGY       =      -1967.20105 KCAL/MOL
          H - H CORRECTION ENERGY =       1627.73479 KCAL/MOL
          MM CORRECTION FOR >N-   =       -117.30927 KCAL/MOL
          MM CORR. FOR -CO-NH-    =         12.10586 KCAL/MOL
                              SUM =     -14156.30868 KCAL/MOL
```

MOPAC v23.0.2 Linux
```
     SCF FIELD WAS ACHIEVED                                   

          HEAT OF FORMATION       =     -14854.02203 KCAL/MOL =  -62149.22817 KJ/MOL

          TOTAL ENERGY            =   -6884384.79913 KCAL/MOL = ELECTRONIC ENERGY + CORE-CORE REPULSION + SOLVATION ENERGY
          ENERGY OF ATOMS         =    6870322.62590 KCAL/MOL
                              SUM =     -14062.17323 KCAL/MOL
          DISPERSION ENERGY       =      -1967.20304 KCAL/MOL
          H-BOND ENERGY           =       -347.17715 KCAL/MOL
          H - H CORRECTION ENERGY =       1627.73479 KCAL/MOL
          MM CORRECTION FOR >N-   =       -117.30927 KCAL/MOL
          MM CORR. FOR -CO-NH-    =         12.10586 KCAL/MOL
                              SUM =     -14854.02203 KCAL/MOL = FINAL HEAT OF FORMATION
```

**Comment by godotalgorithm** [comment]
I can't help much with an issue like this unless you attach the specific input file that is causing the problem or arrange to send me a copy through some other channel. From my experience, energy differences this large are often caused by an SCF problem because the molecule has a region with poorly-formed bonds that cause the HOMO-LUMO gap to be very small. However, this could also be a new bug, and I can't examine the problem without being able to run this example myself.

**Comment by YGuangye** [comment]
Thank you for your prompt response! Unfortunately, I cannot share my input file due to legal concerns. However, I can reproduce the issue using alternate examples from public data. I will add a comment once I have completed the calculation.

**Comment by YGuangye** [comment]
I made another example of this by preparing input `.mop` file from PDB 
[1iep](https://www.rcsb.org/structure/1iep), which is attatched at the bottom of this comment.  
Any assistance would be greatly appreciated.  

formula:
`Empirical Formula: C1467 H2221 N370 O416 S18  =  4492 atoms`  

keywords:  
`PL PM6-D3H4X MOZYME EPS=78.4 NSPA=60 1SCF LBFGS ITER EIGS GRADIENTS DISP`  

MOPAC2016 version 19.350L  
```
     1SCF WAS SPECIFIED, SO BFGS WAS NOT USED                
     SCF FIELD WAS ACHIEVED                                  
 
          HEAT OF FORMATION       =     -18950.16447 KCAL/MOL =  -79287.48816 KJ/MOL
 
          TOTAL ENERGY            =   -9245386.08139 KCAL/MOL
          ENERGY OF ATOMS         =    9225988.99995 KCAL/MOL
                              SUM =     -19397.08144 KCAL/MOL
          DISPERSION ENERGY       =      -2625.07103 KCAL/MOL
          H - H CORRECTION ENERGY =       2394.76217 KCAL/MOL
          MM CORRECTION FOR >N-   =       -151.55893 KCAL/MOL
          MM CORR. FOR -CO-NH-    =          0.95963 KCAL/MOL
                              SUM =     -18950.16447 KCAL/MOL
 
          TOTAL ENERGY            =    -400918.21317 EV
          ELECTRONIC ENERGY       =  -57473879.43503 EV
          CORE-CORE REPULSION     =   57056056.41575 EV
          DIELECTRIC ENERGY       =       -154.47785 EV
          GRADIENT NORM           =       2601.14971 = 38.81016 PER ATOM
          DIPOLE                  =        286.85300 DEBYE  
          NO. OF FILLED LEVELS    =       6274
          CHARGE ON SYSTEM        =         -5
          MOLECULAR WEIGHT        =      32273.9923
          COSMO AREA              =      11210.94 SQUARE ANGSTROMS
          COSMO VOLUME            =      38919.47 CUBIC ANGSTROMS
```

MOPAC v23.0.2 Linux  
```
     1SCF WAS SPECIFIED, SO BFGS WAS NOT USED                
     SCF FIELD WAS ACHIEVED                                  
 
          HEAT OF FORMATION       =     -19792.68407 KCAL/MOL =  -82812.59015 KJ/MOL
 
          TOTAL ENERGY            =   -9244923.08942 KCAL/MOL = ELECTRONIC ENERGY + CORE-CORE REPULSION + SOLVATION ENERGY
          ENERGY OF ATOMS         =    9225996.17910 KCAL/MOL
                              SUM =     -18926.91032 KCAL/MOL
          DISPERSION ENERGY       =      -2625.07369 KCAL/MOL
          H-BOND ENERGY           =       -484.86292 KCAL/MOL
          H - H CORRECTION ENERGY =       2394.76217 KCAL/MOL
          MM CORRECTION FOR >N-   =       -151.55893 KCAL/MOL
          MM CORR. FOR -CO-NH-    =          0.95963 KCAL/MOL
                              SUM =     -19792.68407 KCAL/MOL = FINAL HEAT OF FORMATION
 
          TOTAL ENERGY            =    -400897.80856 EV
          ELECTRONIC ENERGY       =  -57635886.97075 EV
          CORE-CORE REPULSION     =   57234989.16220 EV
          DIELECTRIC ENERGY       =       -153.13144 EV
          GRADIENT NORM           =       2030.10306          =       30.28993 PER ATOM
          DIPOLE                  =        287.17311 DEBYE  
          NO. OF FILLED LEVELS    =       6274
          CHARGE ON SYSTEM        =         -5
          MOLECULAR WEIGHT        =      32273.9923
          COSMO AREA              =      11229.88 SQUARE ANGSTROMS
          COSMO VOLUME            =      38977.94 CUBIC ANGSTROMS
```

[mopac_input.zip](https://github.com/user-attachments/files/18077868/mopac_input.zip)

**Comment by godotalgorithm** [comment]
I discussed this problem with the original MOPAC developer. He was able to reproduce it with some old versions of MOPAC and recognized it as an old, now-corrected bug.

This bug was introduced when MOZYME switched from a sharp transition to a smooth, "feathered" transition between a smeared electron-electron Coulomb interaction and a point-like interaction as the distance between atoms increases. This change was made to improve the continuity of potential energy surfaces in MOZYME calculations, but it was only originally tested for PM7 and inadvertently caused problems in some of the other semiempirical models that went undetected for a while.

I don't know the precise range of versions affected, but the problem occurred as early as version 19.317 and was not fixed until version 20.288 according to MOPAC's [old maintenance logs](http://openmopac.net/Maintenance.html). This problem was fully resolved before the open-source release of MOPAC and does not affect any of the open-source versions.

**Comment by YGuangye** [comment]
Thank you so much for your swift response. Your assistance has made the issue very clear, and I will review my calculations so far to make corrections .
